crates/forge/src/cmd/test/mod.rs

```diff
@@ -916,18 +921,81 @@ fn last_run_failures(config: &Config) -> Option<regex::Regex> {
     }
 }
 
+/// Load persisted failures as qualified contract/test combinations
+pub fn last_run_failures_qualified(config: &Config) -> Option<Vec<(String, String)>> {
+    match fs::read_to_string(&config.test_failures_file) {
+        Ok(content) => {
+            // Parse qualified pattern format: ContractName_testName or legacy format
+            if content.contains('_') && !content.contains('|') {
+                // Single qualified failure
+                if let Some(pos) = content.rfind('_') {
```

Comment:
I don't think this is correct as tests could have the `test_` name format, so in this case a `ContractName_test_Name` pattern will result in `contract_part` being `ContractName_test` and `test_part` - `Name`

Reply:
Good call.

Created `split_qualified_test_name()` function that looks for _test patterns to properly separate contract and test names, with fallback to original behavior.